apps/aecore/src/aec_spend_tx.erl

```diff
@@ -82,15 +97,27 @@ nonce(#spend_tx{nonce = Nonce}) ->
 
 -spec origin(tx()) -> aec_keys:pubkey().
 origin(#spend_tx{} = Tx) ->
-    sender(Tx).
+    sender_pubkey(Tx).
 
--spec sender(tx()) -> aec_keys:pubkey().
+-spec sender(tx()) -> aec_id:id().
 sender(#spend_tx{sender = Sender}) ->
+    Sender.
+
+-spec sender_pubkey(tx()) -> aec_keys:pubkey().
+sender_pubkey(#spend_tx{sender = Sender}) ->
     aec_id:specialize(Sender, account).
 
--spec recipient(tx()) -> aec_keys:pubkey().
+-spec recipient(tx()) -> aec_id:id().
 recipient(#spend_tx{recipient = Recipient}) ->
-    aec_id:specialize(Recipient, account).
+    Recipient.
+
+resolve_recipient(#spend_tx{recipient = Recipient}, Trees) ->
+    case aec_id:specialize(Recipient) of
+        {account,  RecipientPubkey} -> {ok, RecipientPubkey};
+        {oracle,   RecipientPubkey} -> {ok, RecipientPubkey};
+        {contract, RecipientPubkey} -> {ok, RecipientPubkey};
+        {name, NameHash} -> aens:resolve_from_hash(account_pubkey, NameHash, aec_trees:ns(Trees))
```

Comment:
Re-using review effort at [now superseded PR](https://github.com/aeternity/epoch/pull/1127/files#diff-89b8ec7c2f28f3e3c7d21ca9bf070bb7):
As in case of name this is not a pubkey but a hash, shall the Swagger param be renamed from [recipient_pubkey](https://github.com/aeternity/epoch/blob/5a72c277eb0f8619d3c7f8af5c94e63263db10df/config/swagger.yaml#L2471).